replica/table.cc

```diff
@@ -2458,6 +2517,34 @@ future<> tablet_storage_group_manager::handle_tablet_split_completion(const loca
     return stop_fut;
 }
 
+future<> tablet_storage_group_manager::merge_completion_fiber() {
+    co_await coroutine::switch_to(_t.get_config().streaming_scheduling_group);
+    auto& cm = _t.get_compaction_manager();
+
+    while (!_t.async_gate().is_closed()) {
+        try {
+            co_await for_each_storage_group_gently([&cm] (storage_group& sg) -> future<> {
+                auto main_group = sg.main_compaction_group();
+                for (auto& group : sg.merging_groups()) {
+                    // Flushes memtable, so all the data can be moved.
+                    co_await group->flush();
```

Comment:
You're right. we must serialize with a pending write that sleeps in dirty_memory_region_group().run_when_memory_available(), inside table::do_apply(). Newer writes aren't problem since they pick a different group (not merging ones).